var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -145,7 +146,10 @@ class Petsc(Package):
     depends_on('metis@5:~int64', when='@3.8:+metis~int64')
     depends_on('metis@5:+int64', when='@3.8:+metis+int64')
 
-    depends_on('hdf5+mpi+hl+fortran', when='+hdf5+mpi')
+    depends_on('hdf5@:1.10.99+mpi+hl+fortran', when='@:3.12.99+hdf5+mpi')
+    depends_on('hdf5@:1.10.99+mpi+hl+fortran', when='@xsdk-0.2.0+hdf5+mpi')
+    depends_on('hdf5+mpi+hl+fortran', when='@3.13:+hdf5+mpi')
+    depends_on('hdf5+mpi+hl+fortran', when='@develop:+hdf5+mpi')
```

Comment:
The second and 4th lines are not needed as spack sorts and handles those versions correctly

Wrt the 3rd line - its probably best to add it after the fix lands in petsc/master branch

Reply:
So what would you like? Just constrain unconditionally?
```
depends_on('hdf5@:1.10.99+mpi+hl+fortran', when='+hdf5+mpi')
```
And you'll fix it up once support is added to your master branch next release? That's just fine with me, I was just trying to address @jedbrown's comment in #15553 by being preemptive about upcoming changes.

Reply:
yes will update to hdf5-1.12 once petsc/master is updated.

petsc/master will continue to work with hdf5-1.10 after the fix anyway [so it won't be broken]

Reply:
Okay, force pushed with the simple solution.